src/server/src/test/java/com/spotify/reaper/unit/resources/ClusterResourceTest.java

```diff
@@ -31,126 +31,169 @@
   @Test
   public void addingAClusterAutomaticallySetupSchedulingRepairsWhenEnabled() throws Exception {
-    when(jmxProxy.getKeyspaces()).thenReturn(Lists.newArrayList("keyspace1"));
-    when(jmxProxy.getTableNamesForKeyspace("keyspace1")).thenReturn(Sets.newHashSet("table1"));
+    final MockObjects mocks = initMocks();
 
-    context.config = TestRepairConfiguration.defaultConfigBuilder()
-        .withAutoScheduling(TestRepairConfiguration.defaultAutoSchedulingConfigBuilder()
-            .thatIsEnabled()
-            .withTimeBeforeFirstSchedule(Duration.ofMinutes(1))
-            .build())
-        .build();
+    when(mocks.jmxProxy.getKeyspaces()).thenReturn(Lists.newArrayList("keyspace1"));
+    when(mocks.jmxProxy.getTableNamesForKeyspace("keyspace1"))
+        .thenReturn(Sets.newHashSet("table1"));
 
-    ClusterResource clusterResource = new ClusterResource(context);
-    Response response = clusterResource.addCluster(uriInfo, Optional.of(SEED_HOST));
+    mocks.context.config =
+        TestRepairConfiguration.defaultConfigBuilder()
+            .withAutoScheduling(
+                TestRepairConfiguration.defaultAutoSchedulingConfigBuilder()
+                    .thatIsEnabled()
+                    .withTimeBeforeFirstSchedule(Duration.ofMinutes(1))
+                    .build())
+            .build();
+
+    ClusterResource clusterResource = new ClusterResource(mocks.context);
+    Response response = clusterResource.addCluster(mocks.uriInfo, Optional.of(SEED_HOST));
 
     assertEquals(201, response.getStatus());
-    assertThat(context.storage.getAllRepairSchedules()).hasSize(1);
-    assertThat(context.storage.getRepairSchedulesForClusterAndKeyspace(CLUSTER_NAME, "keyspace1")).hasSize(1);
+    assertThat(mocks.context.storage.getAllRepairSchedules()).hasSize(1);
+    assertThat(
+            mocks.context.storage.getRepairSchedulesForClusterAndKeyspace(
+                CLUSTER_NAME, "keyspace1"))
+        .hasSize(1);
+  }
+
+  private MockObjects initMocks() throws ReaperException {
+    AppContext context = new AppContext();
+    UriInfo uriInfo;
+    JmxProxy jmxProxy;
+
+    context.storage = new MemoryStorage();
+    context.config = TestRepairConfiguration.defaultConfig();
+
+    uriInfo = mock(UriInfo.class);
+    when(uriInfo.getAbsolutePath()).thenReturn(SAMPLE_URI);
+    when(uriInfo.getBaseUri()).thenReturn(SAMPLE_URI);
+
+    jmxProxy = mock(JmxProxy.class);
+    when(jmxProxy.getClusterName()).thenReturn(CLUSTER_NAME);
+    when(jmxProxy.getPartitioner()).thenReturn(PARTITIONER);
+    when(jmxProxy.getLiveNodes()).thenReturn(Arrays.asList(SEED_HOST));
+    context.jmxConnectionFactory = new JmxConnectionFactory() {
+      @Override
+      public JmxProxy connect(Optional<RepairStatusHandler> handler, String host, int connectionTimeout)
+          throws ReaperException {
+        return jmxProxy;
+      }
+    };
+
+    return new MockObjects(context, uriInfo, jmxProxy);
+
+  }
+
+  private class MockObjects {
```

Comment:
i'm not too sure why this is an instance class? there's no state in the surrounding class to use.

was this meant as `private static class MockObjects {`

Reply:
meant to be `private static class...` indeed. I'm making the change.